### PR TITLE
Html5 inputs, labels

### PR DESCRIPTION
**Pull request by trasher**
Add possibility to set a field as required (add relevant CSS),
Add possibility to set a type (number, ...)
Add possibility to pass any extra attributes
Add a hack for select2 labels
Add labels on computer fields
Add required attribute on tickets fields (other than dropdowns)
Mark template name as required (creating or editing templates)

### CODE REVIEW

File: inc/html.class.php [hunk 1]
```diff
@@ -3185,6 +3185,7 @@ static function showDateTimeField($name, $options = array()) {
       $p['showyear']   = true;
       $p['display']    = true;
       $p['rand']       = mt_rand();
+      $p['required']   = false;
```
Comment (orthagh) [inline inc/html.class.php]: missing dockblock

File: inc/html.class.php [hunk 2]
```diff
@@ -3777,8 +3782,11 @@ static function showToolTip($content, $options=array()) {
     *    - user    : integer / restrict to a defined user (default -1 : no restriction)
     *    - option  : string / options to add to text field
     *    - display : boolean / if false get string
+    *    - type    : string / html5 field type (number, date, text, ...) defaults to 'text'
+    *    - required: boolean / whether the field is required
+    *    - rand    : integer / pre-exsting random value
     *
-    * @return nothing (print out an HTML div)
+    * @return void (print out an HTML div)
```
Comment (orthagh) [inline inc/html.class.php]: depending on display option, the html can be returned (the comment was already bad previously)

File: inc/html.class.php [hunk 3]
```diff
@@ -3796,19 +3804,35 @@ static function autocompletionTextField(CommonDBTM $item, $field, $options=array
...
+         $output .=  "<input ".$params['option']." id='text$name' type='{$params['type']}' name='".
                        $params['name']."' value=\"".self::cleanInputText($params['value'])."\"
-                       class='autocompletion-text-field'>";
+                       class='autocompletion-text-field'";
+
+         if ($params['required'] == true) {
+            $output .= " required='required'";
+         }
+
+         if (isset($params['attrs'])) {
```
Comment (orthagh) [inline inc/html.class.php]: attrs is a new options for autocompletiontextfield, no ? if yes, we need a default entry in $params and a dockblock

Comment (trasher) [inline inc/html.class.php]: OK for the docbloc; but since there is an `isset` on the entry; it should not be mandatory to add it in the `$params`.

Comment (orthagh) [inline inc/html.class.php]: ok indeed

**Pull request merged**